tensorflow/compiler/xla/mlir_hlo/mhlo/transforms/hlo_legalize_to_lhlo/hlo_legalize_to_lhlo.cc

```diff
@@ -91,6 +92,7 @@ Value insertAlloc(Location loc, OpResult result,
   if (!resultType || !resultType.hasStaticShape()) {
     result.getDefiningOp()->emitOpError()
         << "tensor to buffer conversion expects statically shaped results";
+    return NULL;
```

Comment:
We should change the signature of this function to return `FailureOr<Value>` I think, because right now it's not clear to me that the call sites are properly handling the empty value returned here.

Can you also make it so that these code paths have unit-tests?

Reply:
While trying to figure out how to write the tests I found that this fix is unnecessary. `resultType` and `resultType.hasStaticShape` are validated right before calling `insertAlloc()`/`insertDynamicAlloc()`, that means the control is never reached `emitOpError()` function.
So I see two options:

1.  Revert all these changes with `FailureOr<Value>` and remove `resultType` validation in `insertAlloc()`/`insertDynamicAlloc()` functions.
2.  Accept this fix in case there would be new code using `insertAlloc()`/`insertDynamicAlloc()` without preliminary `resultType` validation. I tried to call functions with incorrect type of `result.value()` without security guards and current `emitOpError`-handling leads to segmentation fault.

@joker-eph what do you think?

Reply:
I'd say: revert the changes and add assertions instead of the emitOpError and dynamic checks

Reply:
Done. Should I open PR to tensorflow/mlir_hlo?

Reply:
No need, thanks!